src/docfx/lib/git/GitCommitProvider.cs

```diff
@@ -193,9 +193,12 @@ public Task SaveCache()
                 using (var writer = new BinaryWriter(stream))
                 {
                     // Create a snapshot of commit cache to ensure count and items matches.
-                    var commitCache = _commitCache.ToList();
+                    //
+                    // There is a race condition in Linq ToList() method, use ConcurrentDictionary.ToArray() to create a snapshot
+                    // https://stackoverflow.com/questions/11692389/getting-argument-exception-in-concurrent-dictionary-when-sorting-and-displaying
+                    var commitCache = _commitCache.ToArray();
```

Comment:
I believe what we need know if it's correct if the get/save called parallel, not to only resolving the tech problem

Reply:
It is not intended, but this class is thread safe and support calling it in parallel.

Reply:
loc will use commitProvider.Get(), so i hope it can be considered:
1. cache the commitProvider per repo path
2. always call saveCache() at the end of the build of each commit provider

Reply:
That is a different task. But seems like your PR has covered this so I'll go ahead and close this one.